templates/python/register.go

```diff
@@ -0,0 +1,253 @@
+	"github.com/golang/protobuf/ptypes/timestamp"
+	"github.com/lyft/protoc-gen-star"
+	"github.com/lyft/protoc-gen-star/lang/go"
+	"github.com/envoyproxy/protoc-gen-validate/templates/shared"
+)
+
+
+func Register(tpl *template.Template, params pgs.Parameters) {
+	fns := PythonFuncs{pgsgo.InitContext(params)}
+
+	tpl.Funcs(map[string]interface{}{
+		"accessor": fns.accessor,
+		"err": fns.err,
+		"lit": fns.lit,
+		"output": fns.output,
+		"lookup": fns.lookup,
+		"inKey": fns.inKey,
+		"byteStr": fns.byteStr,
+		"typ": fns.Type,
+		"oneof": fns.oneofTypeName,
+		"enumValues": fns.enumValues,
+		"name": fns.Name,
+		"externalEnums": fns.externalEnums,
+		"enumPackages": fns.enumPackages,
+		"durLit": fns.durLit,
+		"durStr": fns.durStr,
+		"durGt": fns.durGt,
+		"tsLit": fns.tsLit,
+		"tsStr": fns.tsStr,
+		"tsGt": fns.tsGt,
+	})
+
+	template.Must(tpl.Parse(fileTpl))
+	template.Must(tpl.New("msg").Parse(msgTpl))
+	template.Must(tpl.New("const").Parse(constTpl))
+	template.Must(tpl.New("ltgt").Parse(ltgtTpl))
+	template.Must(tpl.New("in").Parse(inTpl))
+	template.Must(tpl.New("required").Parse(requiredTpl))
+
+	template.Must(tpl.New("none").Parse(noneTpl))
+	template.Must(tpl.New("float").Parse(numTpl))
+	template.Must(tpl.New("double").Parse(numTpl))
+	template.Must(tpl.New("int32").Parse(numTpl))
+	template.Must(tpl.New("int64").Parse(numTpl))
+	template.Must(tpl.New("uint32").Parse(numTpl))
+	template.Must(tpl.New("uint64").Parse(numTpl))
+	template.Must(tpl.New("sint32").Parse(numTpl))
+	template.Must(tpl.New("sint64").Parse(numTpl))
+	template.Must(tpl.New("fixed32").Parse(numTpl))
+	template.Must(tpl.New("fixed64").Parse(numTpl))
+	template.Must(tpl.New("sfixed32").Parse(numTpl))
+	template.Must(tpl.New("sfixed64").Parse(numTpl))
+
+	template.Must(tpl.New("bool").Parse(constTpl))
+	template.Must(tpl.New("string").Parse(strTpl))
+	template.Must(tpl.New("bytes").Parse(bytesTpl))
+
+	template.Must(tpl.New("email").Parse(emailTpl))
+	template.Must(tpl.New("hostname").Parse(hostTpl))
+
+	template.Must(tpl.New("enum").Parse(enumTpl))
+	template.Must(tpl.New("message").Parse(messageTpl))
+	template.Must(tpl.New("repeated").Parse(repTpl))
+	template.Must(tpl.New("map").Parse(mapTpl))
+
+	template.Must(tpl.New("any").Parse(anyTpl))
+	template.Must(tpl.New("duration").Parse(durationTpl))
+	template.Must(tpl.New("durationcmpTpl").Parse(durationcmpTpl))
+
+
+	template.Must(tpl.New("timestampcmp").Parse(timestampcmp))
+	template.Must(tpl.New("timestamp").Parse(timestampTpl))
+
+	template.Must(tpl.New("wrapper").Parse(wrapperTpl))
+}
+
+type PythonFuncs struct{ pgsgo.Context }
+
+
+func (fns PythonFuncs) accessor(ctx shared.RuleContext, reason ...interface{}) string {
+	if ctx.AccessorOverride != "" {
+		return ctx.AccessorOverride
+	}
+	return fmt.Sprintf("m.%s", ctx.Field.Name())
+}
+
+func (fns PythonFuncs) err(ctx shared.RuleContext, reason ...interface{}) string {
+	return fmt.Sprintf("return False, '''%s'''", fmt.Sprint(reason))
```

Comment:
Should this do some joining by "` `" or something? Right now if no reason is provided, this just prints "`[]`"

Reply:
Reading more of the error messages, I think just joining by empty string would be sufficient.